lib/payload.json

```diff
@@ -34,6 +51,40 @@
           "format": "date-time",
           "description": "Date and time of the measurement (RFC3339)"
         },
+        "soil": {
+          "type": "object",
+          "properties": {
+            "moisture": {
+              "description": "Soil moisture (%)",
+              "$ref": "#/definitions/percentage"
+            },
+            "temperature": {
+              "description": "Soil temperature (°C)",
+              "$ref": "#/definitions/temperature"
+            },
+            "EC": {
```

Comment:
Keys need to be lowerCamelCase, and if the first part is an abbreviation then it's lowercase; `ec`.

(otherwise uppercase, like `httpClient` vs `newHTTPClient`)

Same for `n`, `p` and `k` below.